arcgis-ios-sdk-samples/Analysis/Viewshed (GeoElement)/ViewshedGeoElementViewController.swift

```diff
@@ -0,0 +1,136 @@
+                                                       maxDistance: 250.0,
+                                                       headingOffset: 0.0,
+                                                       pitchOffset: 0.0)
+        
+        // offset viewshed observer location to top of tank
+        geoElementViewshed.offsetZ = 3.0
+        
+        // create an analysis overlay to add the viewshed to the scene view
+        let analysisOverlay = AGSAnalysisOverlay()
+        analysisOverlay.analyses.add(geoElementViewshed)
+        sceneView.analysisOverlays.add(analysisOverlay)
+
+        // set camera controller to follow tank
+        let cameraController = AGSOrbitGeoElementCameraController(targetGeoElement: tank, distance: 200.0)
+        cameraController.cameraPitchOffset = 45.0
+        sceneView.cameraController = cameraController
+        
+        sceneView.scene = scene
+    }
+
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // set the new waypoint
+        waypoint = mapPoint;
+        
+        // start a new timer
+        timer = Timer.scheduledTimer(timeInterval: 0.1,
+                                     target:self as Any,
+                                     selector: #selector(animate),
+                                     userInfo: nil,
+                                     repeats: true)
+    }
+    
+    func animate() {
+        guard let waypoint = waypoint,
+            let location = tank.geometry as? AGSPoint else { return }
+        
+        guard let distance = AGSGeometryEngine.geodeticDistanceBetweenPoint1(location,
+                                                                       point2: waypoint,
+                                                                       distanceUnit: AGSLinearUnit.meters(),
+                                                                       azimuthUnit: AGSAngularUnit.degrees(),
+                                                                       curveType: .geodesic) else { return }
+
+        // move toward waypoint a short distance
+        let locations = AGSGeometryEngine.geodeticMove([location],
+                                                       distance: 1.0,
+                                                       distanceUnit: AGSLinearUnit.meters(),
+                                                       azimuth: distance.azimuth1,
+                                                       azimuthUnit: AGSAngularUnit.degrees(),
+                                                       curveType: .geodesic)
+        
+        if let newLocation = locations?.first {
+            tank.geometry = newLocation
+        }
+        
+        if let heading = tank.attributes["HEADING"] as? Double {
+            tank.attributes["HEADING"] = heading + ((distance.azimuth1 - heading) / 10)
+        }
+        
+        if distance.distance <= 5 {
```

Comment:
`distance.distance` feels a little awkward. Should be this be `result.distance`? or something like `distanceResult.distance`? Also, be good to add a comment saying we are stopping the animation once we get within 5 meters of the way point (i think that is what this is about)